controllers/articles.controller.js

```diff
@@ -2,9 +2,12 @@ const articlesModel = require("../models/articles.model");
 const usersModel = require("../models/users.model");
 
 exports.getArticles = (req, res, next) => {
-  articlesModel
-    .getArticlesModel()
-    .then((articles) => {
+  const { sort_by, order, topic } = req.query;
+  Promise.all([
+    articlesModel.checkArticleQueryExists(req.query),
+    articlesModel.getArticlesModel(sort_by, order, topic),
+  ])
+    .then(([bool, articles]) => {
       res.status(200).send({ articles });
     })
     .catch((err) => {
```

Comment:
for a little syntatic-sugar - you can write the catch block like this `.catch(next);` and express will read that the same way - a nice little shorthand